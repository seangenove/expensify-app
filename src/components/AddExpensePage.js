import React from 'react';

const AddExpensePage = () => (
    <div>
        This is my add expense component
    </div>
);

export default AddExpensePage;