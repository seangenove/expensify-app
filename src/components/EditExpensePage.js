import React from 'react';

// const EditExpensePage = () => (
//     <div>
//         // This is my edit add expense component
//     </div>
// );

const EditExpensePage = (props) => {
    console.log(props);

    return (
        <div>
        Editing the expense with id of {props.match.params.id}
        </div>
    );
};

export default EditExpensePage;