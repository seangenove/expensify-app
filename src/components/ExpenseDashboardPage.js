import React from 'react';
import ExpenseList from './ExpenseList';

const ExpenseDashboardPage = () => (
    <div>
        This is from my dashboard component
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;