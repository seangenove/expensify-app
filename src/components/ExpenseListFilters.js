import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByAmount, sortByDate } from './../actions/filters'

const ExpenseListFilters = (props) => (
    <div>
        <input
            type="text"
            value={props.filters.text}
            onChange={(e) => {
                props.dispatch(setTextFilter(e.target.value))
                console.log(e.target.value);
            }}
        />
        <select
            onChange={(e) => {
                if (e.target.value === 'date') {
                    props.dispatch(sortByDate())
                } else if (e.target.value === 'amount') {
                    props.dispatch(sortByAmount())
                }
                console.log(e.target.value)
            }}
        >
            <option value="date" selected={props.filters.sortBy === 'date' ? "selected" : null}>
                Date
            </option>
            <option value="amount" selected={props.filters.sortBy === 'amount' ? "selected" : null}>
                Amount
            </option>
        </select>
    </div>
);

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

export default connect(mapStateToProps)(ExpenseListFilters);