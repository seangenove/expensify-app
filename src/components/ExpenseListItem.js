import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

// load a locale
// numeral.register('locale', 'php', {
//     delimiters: {
//         thousands: ',',
//         decimal: '.'
//     },
//     abbreviations: {
//         thousand: 'k',
//         million: 'm',
//         billion: 'b',
//         trillion: 't'
//     },
//     ordinal : function (number) {
//         return number === 1 ? 'er' : 'ème';
//     },
//     currency: {
//         symbol: '₱'
//     }
// });

// numeral.locale('php');

const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>
            {numeral(amount / 100 ).format('$0,0.00')}
            -
            {moment(createdAt).format('MMM Do, YYYY')}
        </p>
    </div>
);

export default ExpenseListItem;
