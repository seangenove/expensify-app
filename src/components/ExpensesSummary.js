import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from './../selectors/expenses'
import getExpensesTotal from './../selectors/expenses-total'
import numeral from 'numeral';

// numeral.locale('php');

export const ExpensesSummary = ({ visibleExpenseCount, visibleExpenseSum }) => {
    const expenseWord = visibleExpenseCount === 1 ? 'expense' : 'expenses'
    const formattedExpenseTotal = numeral(visibleExpenseSum / 100).format('$0,0.00');


    return (
        <div>
            <h3>
                Viewing {visibleExpenseCount} {expenseWord} totalling to {formattedExpenseTotal}
            </h3>
        </div>)
}

const mapStateToProps = (state) => {
    const visibleExpenseCount = selectExpenses(state.expenses, state.filters).length

    return {
        visibleExpenseCount: visibleExpenseCount,
        visibleExpenseSum: getExpensesTotal(selectExpenses(state.expenses, state.filters))
    }
};

export default connect(mapStateToProps)(ExpensesSummary);

