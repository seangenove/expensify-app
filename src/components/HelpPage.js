import React from 'react';

const HelpPage = () => (
    <div>
        This is my help component
    </div>
);

export default HelpPage;