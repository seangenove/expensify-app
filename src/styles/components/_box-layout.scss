.box-layout {
    align-items: center;
    background: url('/images/bg.jpg');
    background-size: cover;
    display: flex;
    justify-content: center;
    height: 100vh;
    width: 100vw;
}

.box-layout__box {
    background: fade-out($color: white, $amount: .15);
    border-radius: 3px;
    padding: $l-size $m-size;
    text-align: center;
    width: 25rem;
}

.box-layout__title {
    margin: 0 0 $xs-size 0;
    line-height: 1;
}
.box-layout__sub-title {
    color: lighten($grey, 20%);
    line-height: 1;
    margin: 0 0 $m-size 0;
    font-size: $s-size;
    
}