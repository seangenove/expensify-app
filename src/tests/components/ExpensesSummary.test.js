import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from './../../components/ExpensesSummary';
import expenses from './../fixtures/expenses';

test('Should render ExpensesSummary correctly with 1 expense', () => {
    const wrapper = shallow(
        <ExpensesSummary
            visibleExpenseCount={1}
            visibleExpenseSum={500}
        />
    );

    expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesSummary correctly with multiple expenses', () => {
    const wrapper = shallow(
        <ExpensesSummary
            visibleExpenseCount={23}
            visibleExpenseSum={69000}
        />
    );
    
    expect(wrapper).toMatchSnapshot();
});
