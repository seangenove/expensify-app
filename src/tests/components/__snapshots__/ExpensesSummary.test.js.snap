// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary correctly with 1 expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expense
     totalling to 
    $5.00
  </h3>
</div>
`;

exports[`Should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h3>
    Viewing 
    23
     
    expenses
     totalling to 
    $690.00
  </h3>
</div>
`;
